### PR TITLE
[Bug] fix `replace` function bug

### LINKED ISSUE #6604: [Bug] `replace` function incorrect
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

trunk version , commit_id = `2c58724c84021ae5c9fdd5ba304808b1e79191d4`

### What's Wrong?

The `replace` function cannot replace corecctly when the old substring is at the beginning of the original string.

```sql 
MySQL [(none)]>  select replace('http://www.baidu.com:9090','http', 'ftp');
+-----------------------------------------------------+
| replace('http://www.baidu.com:9090', 'http', 'ftp') |
+-----------------------------------------------------+
| http://www.baidu.com:9090                           |
+-----------------------------------------------------+
1 row in set (0.034 sec)
```

### What You Expected?

The correct result is :

```sql 
mysql> select replace('http://www.baidu.com:9090','http', 'ftp');
+-----------------------------------------------------+
| replace('http://www.baidu.com:9090', 'http', 'ftp') |
+-----------------------------------------------------+
| ftp://www.baidu.com:9090                            |
+-----------------------------------------------------+
1 row in set (0.02 sec)
```

### How to Reproduce?

```sql 
MySQL [(none)]>  select replace('http://www.baidu.com:9090','http', 'ftp');
+-----------------------------------------------------+
| replace('http://www.baidu.com:9090', 'http', 'ftp') |
+-----------------------------------------------------+
| http://www.baidu.com:9090                           |
+-----------------------------------------------------+
1 row in set (0.034 sec)
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qidaye**
## Proposed changes

fix #6604 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6604 ) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

File: be/test/exprs/string_functions_test.cpp [hunk 1]
```diff
@@ -559,6 +559,11 @@ TEST_F(StringFunctionsTest, replace) {
     ASSERT_EQ(StringVal("http://华夏zhongguo:9090"),
               StringFunctions::replace(ctx, StringVal("http://中国hello:9090"),
                                        StringVal("中国hello"), StringVal("华夏zhongguo")));
+
+    //old substring is at the beginning of string
+    ASSERT_EQ(StringVal("ftp://www.baidu.com:9090"),
+              StringFunctions::replace(ctx, StringVal("http://www.baidu.com:9090"),
+                                       StringVal("http"), StringVal("ftp")));
```
Comment (yangzhg) [inline be/test/exprs/string_functions_test.cpp:566]: better add an empty string case

Comment (qidaye) [reply be/test/exprs/string_functions_test.cpp:566]: There already has a empty case in above.

**Review by yangzhg — DISMISSED** [review]
LGTM

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**